torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -180,34 +181,24 @@ def aot_dispatch_base(
     if not hasattr(compiled_fw, "_boxed_call"):
         compiled_fw = make_boxed_func(compiled_fw)
 
-    # Create a wrapper to set up the rng functionalize bits
-    @wraps(compiled_fw)
-    def rng_functionalization_wrapper(args: List[Any]):
-        # see note: [Returning Fake Tensors on First AOT Autograd Call]
-        nonlocal fakified_out
-        if fakified_out is not None:
-            out = fakified_out
-            fakified_out = None
-            return out
-
-        if fw_metadata.is_rng_op_functionalized:
-            # Add the seed and offset to args
-            seed, offset = CUDARngStateHelper.get_torch_state_as_tuple()
-            args.extend([seed, offset])
-            out = compiled_fw(args)
-            out = functionalized_rng_runtime_epilogue(fw_metadata, out)
-            return out
-        else:
-            return compiled_fw(args)
-
-    if maybe_subclass_meta is not None:
-        compiled_fw_func = aot_dispatch_subclass_wrapper(
-            rng_functionalization_wrapper,
-            subclass_metas=fw_metadata.subclass_fw_graph_out_meta,
-            num_fw_outs_saved_for_bw=None,
-        )
-    else:
-        compiled_fw_func = rng_functionalization_wrapper
+    # Create a wrapper to set up the rng functionalize and fakified out bits
+    compiled_fw = FunctionalizedRngRuntimeWrapper().post_compile(
```

Comment:
cc @anijain2305. James is refactoring a lot of the runtime wrappers in AOTAutograd - I grepped around and I don't think the config that turns on rng functionalization inside of AOTAutograd is actually used anywhere. Do you think we want to keep that logic around longer term? Or should we try to kill it.